README.md

```diff
@@ -149,6 +149,68 @@ We use this branching model in fuse-starter-java:  http://nvie.com/posts/a-succe
  - Search for the Organization 'Galatea'
  - Bind to 'starter-java' and accept
 
+
+## Spring Annotations:
+Annotations are used to provide supplemental information about a program. They allow you to direct the framework to do your bidding, taking control and overriding its defaults when needed.
+Annotations are quick, easy to use, and orders of magnitude faster than building the equivalent functionality yourself.
+
+For a detailed list of annotations check (https://lightrun.com/java/the-complete-list-of-spring-boot-annotations-you-must-know/)
+
+
+- Basic setup:
+  - @Bean:
+    - It is a method-level annotation. It is an alternative of XML <bean> tag. It tells the method to produce a bean to be managed by Spring Container.
+  - @Configuration:
+    - Mark a configuration class that declares one or more @Bean methods
+  - @ComponentScan:
+    - It is used when we want to scan a package for beans.
+  - @EnableAutoConfiguration:
+    - It auto-configures the bean that is present in the classpath and configures it to run the methods.
+  - @SpringBootApplication:
+    - It is a combination of three annotations @EnableAutoConfiguration, @ComponentScan, and @Configuration
+
+  - https://www.javatpoint.com/spring-boot-annotations#:~:text=%7D%C2%A0%C2%A0-,Spring,-Boot%20Annotations for more information
+
+
+- Stereotype Annotations:
+  - @Component: class-level annotation
+  - @Service
+  - @Repository
+  - @Controller
+  - @RestController
+
+  - Allows Spring to automatically detect our custom beans. (For more visit https://www.baeldung.com/spring-component-annotation#component)
```

Comment:
In particular, these are class-level annotations that will result in spring creating an instance of the class as a bean without us needing to define the bean in a configuration class